bakery/src/scripts/disassemble-book.py

```diff
@@ -51,28 +51,34 @@ def main():
         # without diverging
         slugs = extract_slugs_from_binder(binder)
 
+    with open(baked_metdata_file, "r") as baked_json:
+        baked_metadata = json.load(baked_json)
+        book_toc_metadata = baked_metadata.get(binder.id)
+
     nav = html_root.xpath("//xhtml:nav", namespaces=HTML_DOCUMENT_NAMESPACES)[0]
 
     toc_maker = ElementMaker(namespace=None,
                              nsmap={None: "http://www.w3.org/1999/xhtml"})
     toc = toc_maker.html(E.head(E.title("Table of Contents")),
                          E.body(nav))
 
-    with open(f"{out_dir}/collection.toc.xhtml", "wb") as out:
-        out.write(etree.tostring(toc, encoding="utf8", pretty_print=True))
+    nav_links = toc.xpath("//xhtml:a", namespaces=HTML_DOCUMENT_NAMESPACES)
 
-    with open(baked_metdata_file, "r") as baked_json:
-        baked_metadata = json.load(baked_json)
-        book_toc_metadata = baked_metadata.get(collection_id)
+    for doc in flatten_to_documents(binder):
+        id_with_context = f'{collection_uuid}@{collection_version}:{doc.ident_hash}'
 
-    with open(f"{out_dir}/collection.toc-metadata.json", "w") as toc_json:
-        json.dump(book_toc_metadata, toc_json)
+        module_etree = content_to_etree(doc.content)
+        for link in nav_links:
+            link_href = link.attrib['href']
+            if not link_href.startswith('#'):
+                continue
+            if module_etree.xpath(f"/xhtml:body/xhtml:div[@id='{link_href[1:]}']", namespaces=HTML_DOCUMENT_NAMESPACES):
```

Comment:
This assumes that the table of contents links to the `page` or `composite-page` element. Anyone making recipes would need to be aware of this (although this is how the table of contents is generated already - there is a release flag that adds composte-page links to the table of contents in this manner as well. i.e. not all books will have their links work right away - only the ones in this list https://github.com/openstax/cnx-recipes/blob/master/recipes/books/_release.scss#L232)